### PR TITLE
code clean up, remove few todos and little optimisations

### PR DESCRIPTION
**Pull request by sogunshola**
---
name: 'Pull request'
about: A new pull request
---

### Checklist

- [X] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [X] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [X] I have made corresponding changes to the documentation
- [X] I have updated Typescript definitions when needed

### Issue Description

Related issue: #`FILL_THIS_OUT`

### CODE REVIEW

File: packages/evmApi/src/resolvers/PaginatedResolver.ts [hunk 1]
```diff
@@ -92,14 +92,12 @@ export class EvmPaginatedResolver<
...
-    const bodyParams = this.getBodyParams(apiParams);
-
     const { result } = await RequestController.post<
       ServerResponse<PaginatedResponse<ApiResult>>,
       Record<string, string>,
-      Record<string, string>
-    >(url, searchParams, bodyParams);
+      //@ts-ignore TODO: fix the ApiParams type, as it should extend object/record
+      ApiParams
+    >(url, {}, apiParams);
```
Comment (ErnoW) [inline packages/evmApi/src/resolvers/PaginatedResolver.ts:100]: Why did you remove the searchParams? 
We can have post requests that have both search params and bodyParams.

File: packages/evmApi/src/resolvers/Resolver.ts [hunk 2]
```diff
@@ -159,14 +159,12 @@ export class EvmResolver<ApiParams, Params, ApiResult, AdaptedResult, JSONResult
...
 
-    const searchParams = this.getSearchParams(apiParams);
-    const bodyParams = this.getBodyParams(apiParams);
-
-    const { result } = await RequestController.post<
-      ServerResponse<ApiResult>,
-      Record<string, string>,
-      Record<string, string>
-    >(url, searchParams, bodyParams);
+    //@ts-ignore TODO: fix the ApiParams type, as it should extend object/record
```
Comment (ErnoW) [inline packages/evmApi/src/resolvers/Resolver.ts:162]: Why did you remove the searchParams?
We can have post requests that have both search params and bodyParams.

File: packages/evmApi/src/resolvers/defi/getPairAddress.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export const getPairAddressResolver = new EvmResolver({
   name: 'getPairAddress',
   getPath: (params: Params) => `${params.token0Address}/${params.token1Address}/reserves`,
   apiToResult: (data: ApiResult, params: Params) => ({
-    //   ApiResult types generated all come as undefined which should not be the case TODO:
+    //   ApiResult types generated all come as undefined which should not be the case
```
Comment (ErnoW) [inline packages/evmApi/src/resolvers/defi/getPairAddress.ts:25]: Please leave this TODO as it is something that is not addressed yet.

File: packages/evmApi/src/resolvers/token/getAllTokenIds.ts [hunk 4]
```diff
@@ -15,11 +15,30 @@ export interface Params extends Camelize<Omit<ApiParams, 'chain' | 'address'>>,
 }
 
 type ApiResult = operations[operation]['responses']['200']['content']['application/json'];
+interface Result extends Omit<ApiResult, 'result'> {
```
Comment (ErnoW) [inline packages/evmApi/src/resolvers/token/getAllTokenIds.ts:18]: Can you add a TODO to revisit this when api has updated the definition. Also add an issue ticket for this, so we can pick this up next sprint.

Also do not name this Result but rather use ApiResult as it might cause confusion that this is a parsed result.

You can name the original `ApiResult` to `GeneratedApiResult`

File: packages/evmApi/src/resolvers/token/getTokenIdMetadata.ts [hunk 5]
```diff
@@ -17,10 +17,17 @@ export interface Params extends Camelize<Omit<ApiParams, 'chain' | 'address'>> {
 
 type ApiResult = operations[operation]['responses']['200']['content']['application/json'];
 
+export interface Result extends ApiResult {
```
Comment (ErnoW) [inline packages/evmApi/src/resolvers/token/getTokenIdMetadata.ts:20]: Can you add a TODO to revisit this when api has updated the definition. Also add an issue ticket for this, so we can pick this up next sprint.

Also do not name this Result but rather use ApiResult as it might cause confusion that this is a parsed result.

You can name the original `ApiResult` to `GeneratedApiResult`

File: packages/server/src/utils/parseFile.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { File, FullOptions } from 'parse';
```
Comment (ErnoW) [inline packages/server/src/utils/parseFile.ts:1]: Why is this file included?

**Review by ErnoW — CHANGES_REQUESTED** [review]

File: packages/evmApi/src/resolvers/Resolver.ts (hunk 2 above)
Comment (sogunshola) [reply packages/evmApi/src/resolvers/Resolver.ts:162]: I only made this update for server requests. The server requests are all POST and the parameters are all expected in the body.

File: packages/evmApi/src/resolvers/PaginatedResolver.ts (hunk 1 above)
Comment (sogunshola) [reply packages/evmApi/src/resolvers/PaginatedResolver.ts:100]: I only made this update for server requests. The server requests are all POST and the parameters are all expected in the body.

File: packages/server/src/utils/parseFile.ts (hunk 6 above)
Comment (sogunshola) [reply packages/server/src/utils/parseFile.ts:1]: taking out, was not meant to be part of the PR
